pkg/checker/checker.go

```diff
@@ -1,30 +1,57 @@
 package checker
 
 import (
+	"encoding/json"
 	"fmt"
 	"regexp"
 
 	"github.com/hashicorp/go-version"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	remoteauth "oras.land/oras-go/v2/registry/remote/auth"
+
+	"kcl-lang.io/kpm/pkg/constants"
+	"kcl-lang.io/kpm/pkg/downloader"
+	"kcl-lang.io/kpm/pkg/oci"
+	"kcl-lang.io/kpm/pkg/opt"
 	pkg "kcl-lang.io/kpm/pkg/package"
+	"kcl-lang.io/kpm/pkg/reporter"
+	"kcl-lang.io/kpm/pkg/settings"
+	"kcl-lang.io/kpm/pkg/utils"
 )
 
-// Checker defines an interface for KclPkg dependencies checkers.
+// Checker defines an interface for checking KclPkg dependencies.
 type Checker interface {
 	Check(pkg.KclPkg) error
 }
 
+// DepChecker is responsible for running multiple checkers on a package's dependencies.
 type DepChecker struct {
 	checkers []Checker
 }
 
-// NewDepChecker creates a new DepChecker with provided checkers.
-func NewDepChecker(checkers ...Checker) *DepChecker {
-	return &DepChecker{checkers: checkers}
+// DepCheckerOption configures how we set up DepChecker.
+type DepCheckerOption func(*DepChecker)
+
+// NewDepChecker creates a new DepChecker with options.
+func NewDepChecker(options ...DepCheckerOption) *DepChecker {
+	depChecker := &DepChecker{}
+	for _, opt := range options {
+		opt(depChecker)
+	}
+	return depChecker
+}
+
+// WithChecker adds a new Checker to DepChecker.
+func WithChecker(checker Checker) DepCheckerOption {
```

Comment:
Is WithCheckers more appropriate ?

Reply:
Since the intent was to add checkers to `DepChecker` one by one, I named it `WithChecker` to emphasise that it includes a single checker at each step. Let me know if you think another name would be more suitable!

Reply:
Try adding a `WithCheckers` method, one array at a time.